packages/@react-spectrum/table/src/DragPreview.tsx

```diff
@@ -0,0 +1,132 @@
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+import {classNames} from '@react-spectrum/utils';
+import {Direction} from '@react-types/shared';
+import {GridNode} from '@react-types/grid';
+import React from 'react';
+import {SpectrumTableProps} from '@react-types/table';
+import styles from '@adobe/spectrum-css-temp/components/table/vars.css';
+import stylesOverrides from './table.css';
+import type {TableLayout} from '@react-stately/layout';
+
+interface DragPreviewProps<T> {
+  item: GridNode<any>,
+  itemCount: number,
+  density: SpectrumTableProps<T>['density'],
+  layout: TableLayout<T>,
+  direction: Direction
+}
+
+interface TableCellPreviewProps {
+  cell: GridNode<any>,
+  layout: TableLayout<any>,
+  selectionCellWidth: number,
+  dragButtonCellWidth: number,
+  direction: Direction
+}
+
+export function TableCellPreview(props: TableCellPreviewProps) {
+  let {cell, layout, selectionCellWidth, dragButtonCellWidth, direction} = props;
+  let {align} = cell.column.props;
+  let {x: left, height, width} = layout.getLayoutInfo(cell.key).rect;
+  if (cell.props.isSelectionCell) {
+    return null;
+  }
+  // Move cell over since we aren't showing selection or drag cells in preview.
+  left = left - selectionCellWidth - dragButtonCellWidth;
+  let right: number;
+  if (direction === 'rtl') {
+    right = left;
+    left = undefined;
+  }
+  return (
+    <div
+      style={{left, right, height, width, position: 'absolute'}}
+      className={
+          classNames(
+            styles,
+            'spectrum-Table-cell',
+            classNames(
+              stylesOverrides,
+              'react-spectrum-Table-cell',
+              {
+                'react-spectrum-Table-cell--alignStart': align === 'start',
+                'react-spectrum-Table-cell--alignCenter': align === 'center',
+                'react-spectrum-Table-cell--alignEnd': align === 'end'
+              }
+            )
+          )
+        }>
+      <span
+        className={
+            classNames(
+              styles,
+              'spectrum-Table-cellContents'
+            )
+        }>
+        {cell.rendered}
+      </span>
+    </div>
+  );
+}
+
+export function DragPreview(props: DragPreviewProps<unknown>) {
+  let {
+    item,
+    itemCount,
+    density,
+    layout,
+    direction
+  } = props;
+  let {height, width} = layout.getLayoutInfo(item.key).rect;
+
+  let isDraggingMultiple = itemCount > 1;
+
+  let cells = [...item.childNodes];
+  let dragButtonCellWidth = cells[0].props.isDragButtonCell ? layout.getLayoutInfo(cells[0].key).rect.width : 0;
+  let selectionCellWidth = cells[1].props.isSelectionCell ? layout.getLayoutInfo(cells[1].key).rect.width : 0;
+
+  width = width - selectionCellWidth - dragButtonCellWidth;
```

Comment:
Something is off about the Drag Preview position, if I drag from where the drag button would be, the preview is rendered floating off to the side

Reply:
should be fixed now